### DISCUSSION TITLE
Filter for default dashboard widgets

### DISCUSSION THREAD

**Discussion by litebito** [netbox-community/netbox#15130] [Help Wanted!]
Hi,

I've searched around, but either I missed it or it has been asked before. I also could not find an answer immediately in the Netbox docs.
I already created our DEFAULT_DASHBOARD in configuration.py
I would like to create some custom widgets, but filtered or sorted.

For example:

```

    {
        'widget': 'extras.ObjectListWidget',
        'width': 12,
        'height': 5,
        'title': 'Racks',
        'color': 'purple',
        'config': {
            'model': 'dcim.rack',
            'page_size': 25,
        }
    },   
```

I would like to create a widget, containing only the racks with a specific role, and sorted in order of last updated (last updated rack first)
I assume I need to add the filter and/or sorting in the 'config': { } section, but what is the syntax? 

Thanks

**Comment by candlerb** [comment]
https://github.com/netbox-community/netbox/blob/v3.7.2/netbox/extras/dashboard/widgets.py#L215-L228

From that, I would guess (untested) that you add `url_params` in the same format as the [REST API filtering](https://docs.netbox.dev/en/stable/reference/filtering/) would use, in a form suitable for `urlencode()`, e.g.

```
        'config': {
            'model': 'dcim.rack',
            'page_size': 25,
            'url_params': { 'role': 'my_role_slug', 'ordering': '-last_updated' },
        }
```

Quick check of the encoding:

```
>>> from urllib.parse import urlencode
>>> urlencode({ 'role': 'my_role_slug', 'ordering': '-last_updated' })
'role=my_role_slug&ordering=-last_updated'
```

**Comment by litebito** [comment]
Thanks for the fast feedback.
I tried a few variations, and found something interesting: the filtering works, but the ordering on `last_updated` does not seem to work (not ascending or descending).
I'll keep trying/playing around with other combinations now that I know the syntax.

**Reply by candlerb** [reply]
Interesting. Works for me with the API (Netbox v3.7.2):

```
curl -sS -H "Authorization: Token $TOKEN" "$NETBOX/api/dcim/racks/?ordering=last_updated" | jq '.results[].name'
curl -sS -H "Authorization: Token $TOKEN" "$NETBOX/api/dcim/racks/?ordering=-last_updated" | jq '.results[].name'
```

I don't know why the widget doesn't honour this.
